graylog2-web-interface/src/components/event-definitions/event-definitions/EventDefinitionsContainer.jsx

```diff
@@ -56,46 +69,109 @@ class EventDefinitionsContainer extends React.Component {
   };
 
   handleDelete = (definition) => {
-    return () => {
-      if (window.confirm(`Are you sure you want to delete "${definition.title}"?`)) {
-        EventDefinitionsActions.delete(definition);
-      }
-    };
+    this.setState({ showDeleteDialog: true, currentDefinition: definition });
+  };
+
+  handleCopy = (definition) => {
+    this.setState({ showCopyDialog: true, currentDefinition: definition });
   };
 
   handleEnable = (definition) => {
-    return () => {
-      if (window.confirm(`Are you sure you want to enable "${definition.title}"?`)) {
-        EventDefinitionsActions.enable(definition);
-      }
-    };
+    this.setState({ showEnableDialog: true, currentDefinition: definition });
   };
 
   handleDisable = (definition) => {
-    return () => {
-      if (window.confirm(`Are you sure you want to disable "${definition.title}"?`)) {
-        EventDefinitionsActions.disable(definition);
-      }
-    };
+    this.setState({ showDisableDialog: true, currentDefinition: definition });
+  };
+
+  handleClearState = () => {
+    this.setState({
+      showCopyDialog: false,
+      showDeleteDialog: false,
+      showDisableDialog: false,
+      showEnableDialog: false,
+      currentDefinition: null,
+    });
   };
 
   render() {
     const { eventDefinitions } = this.props;
 
+    const {
+      currentDefinition,
+      showCopyDialog,
+      showDeleteDialog,
+      showDisableDialog,
+      showEnableDialog,
+    } = this.state;
+
     if (!eventDefinitions.eventDefinitions) {
       return <Spinner text="Loading Event Definitions information..." />;
     }
 
     return (
-      <EventDefinitions eventDefinitions={eventDefinitions.eventDefinitions}
-                        context={eventDefinitions.context}
-                        pagination={eventDefinitions.pagination}
-                        query={eventDefinitions.query}
-                        onPageChange={this.handlePageChange}
-                        onQueryChange={this.handleQueryChange}
-                        onDelete={this.handleDelete}
-                        onEnable={this.handleEnable}
-                        onDisable={this.handleDisable} />
+      <>
+        <EventDefinitions eventDefinitions={eventDefinitions.eventDefinitions}
+                          context={eventDefinitions.context}
+                          pagination={eventDefinitions.pagination}
+                          query={eventDefinitions.query}
+                          onPageChange={this.handlePageChange}
+                          onQueryChange={this.handleQueryChange}
+                          onDelete={this.handleDelete}
+                          onCopy={this.handleCopy}
+                          onEnable={this.handleEnable}
+                          onDisable={this.handleDisable} />
+        {showCopyDialog && (
```

Comment:
Kind of a nitpick here but is there a particular reason we are doing it this way as opposed to using a generic `<ConfirmDialog>` and a switch based on like a state variable like `actionType`, or something like `dialogState: { isDisplayed: true, type: 'copy|delete|disable|enable'`?

I'm only asking because it looks to be quite redundant having to display 4 code blocks with minimal difference between them, but maybe I'm overthinking it.

Reply:
@simonychuang good point. I will try to simplify this. It is definitely more repetitive than I would like.

Reply:
@simonychuang, ok so I condensed this. I still don't love the execution here but because the child components are all using distinct handlers I had to leave some repetitive code in there.